### PR TITLE
fix: Scheme not starting with alphabetic character

### LINKED ISSUE #11: ERROR: FormatException: Scheme not starting with alphabetic character (at character 1)
I have this error when run the discovery "ERROR: FormatException: Scheme not starting with alphabetic character (at character 1)
I/flutter (13258): _urn:schemas-upnp-org:service:AVTransport_control
I/flutter (13258):  - http://192.168.1.2:49152/description.xml"
Example code that I used

````
import 'package:upnp2/upnp.dart';

// NOTE: Open the example directory to see more elaborate examples
main() async {
  final disc = DeviceDiscoverer();
  await disc.start(ipv6: false);
  disc.quickDiscoverClients().listen((client) async {
    try {
      final dev = await client.getDevice();
      print('Found device: ${dev!.friendlyName}: ${dev.url}');
    } catch (e, stack) {
      print('ERROR: $e - ${client.location}');
      print(stack);
    }
  });
}
````

Xml parsed :
`  
```
<root xmlns="urn:schemas-upnp-org:device-1-0">
<specVersion>
<major>1</major>
<minor>0</minor>
</specVersion>
<device>
<deviceType>urn:schemas-upnp-org:device:MediaRenderer:1</deviceType>
<presentationURL>/</presentationURL>
<friendlyName>T14 EXTREME</friendlyName>
<manufacturer>lucky</manufacturer>
<manufacturerURL>http://www.xxxxx.com</manufacturerURL>
<modelDescription>GX Media Renderer</modelDescription>
<modelName>HappyLucky</modelName>
<modelURL>http://www.xxxxx.com</modelURL>
<dlna:X_DLNADOC xmlns:dlna="urn:schemas-dlna-org:device-1-0">DMR-1.50</dlna:X_DLNADOC>
<UDN>uuid:A252BA22-9245-2D32-A189-BA2292922D39</UDN>
<UID>-7056134368720033613</UID>
<serviceList>
<service>
<serviceType>urn:schemas-upnp-org:service:AVTransport:1</serviceType>
<serviceId>urn:upnp-org:serviceId:AVTransport</serviceId>
<SCPDURL>/dlna-vir/dmr/AVTransport_scpd.xml</SCPDURL>
<controlURL>_urn:schemas-upnp-org:service:AVTransport_control</controlURL>
<eventSubURL>_urn:schemas-upnp-org:service:AVTransport_event</eventSubURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:ConnectionManager:1</serviceType>
<serviceId>urn:upnp-org:serviceId:ConnectionManager</serviceId>
<SCPDURL>/dlna-vir/dmr/ConnectionManager_scpd.xml</SCPDURL>
<controlURL>_urn:schemas-upnp-org:service:ConnectionManager_control</controlURL>
<eventSubURL>_urn:schemas-upnp-org:service:ConnectionManager_event</eventSubURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:RenderingControl:1</serviceType>
<serviceId>urn:upnp-org:serviceId:RenderingControl</serviceId>
<SCPDURL>/dlna-vir/dmr/RenderingControl_scpd.xml</SCPDURL>
<controlURL>_urn:schemas-upnp-org:service:RenderingControl_control</controlURL>
<eventSubURL>_urn:schemas-upnp-org:service:RenderingControl_event</eventSubURL>
</service>
</serviceList>
</device>
<URLBase>http://192.168.1.2:49152/</URLBase>
</root>
```
`

### PR DESCRIPTION
**Pull request by pedia**
fix #11, and add unit test

### CODE REVIEW

File: lib/src/service.dart [hunk 1]
```diff
@@ -44,6 +44,13 @@ class ServiceDescription {
     }
   }
 
+  static Uri patchUrl(Uri uri, String path) {
```
Comment (daniel-naegele) [inline lib/src/service.dart:50]: Could you maybe add a comment/docs what this method does? From the name only it is not clear whats happening here.

**Review by daniel-naegele — CHANGES_REQUESTED** [review]

**Comment by daniel-naegele** [comment]
Are there any other places, where this bug with the Uri resolution could appear? Have you checked this?

File: lib/src/service.dart (hunk 1 above)
Comment (pedia) [reply lib/src/service.dart:50]: Yes, comment added. Also add more test.

Please review again.

**Comment by pedia** [comment]
> Are there any other places, where this bug with the Uri resolution could appear? Have you checked this?

Yes. There are other places that used `Uri.resolve`, but they are safe.

**Pull request merged**